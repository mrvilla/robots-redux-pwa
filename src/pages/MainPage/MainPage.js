import React, { Component } from 'react';
import CardList from '../../components/Card/CardList';
import SearchBox from '../../components/Search/SearchBox';
import Scroll from '../../components/Scroll/Scroll';
import ErrorBoundary from '../../components/ErrorBoundary/ErrorBoundary';
import Header from '../../components/Header/Header';
import { robots } from '../../robots';

export class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots()
  }

  filterRobots = () => {
    const { searchField, robots } = this.props;
    return robots.filter(robot => {
      return robot.name.toLowerCase().includes(searchField.toLowerCase());
    })
  }

  render() {
    const { onSearchChange, isPending } = this.props;

    return (
      <>
        <Header />
        <div  className='tc mb4'>
          <SearchBox
            searchChange={onSearchChange}
          />
        </div>
        <Scroll>
          { isPending ? <h1>Loading....</h1> :
            <ErrorBoundary>
              <CardList
                robots={this.filterRobots(robots)}
              />
            </ErrorBoundary>
          }
        </Scroll>
      </>
    );
  };
}

export default MainPage;
