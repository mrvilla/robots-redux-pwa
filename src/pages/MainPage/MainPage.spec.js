import React from "react";
import renderer from "react-test-renderer";
import MainPage from './MainPage';

describe("MainPage Snapshot testing", () => {
  let domTree;
  beforeEach(() => {
    const mockProps = {
      onRequestRobots: jest.fn(),
      robots: [],
      searchField: '',
      isPending: false
    }
    domTree = renderer.create(<MainPage {...mockProps} />);
  })

  it("MainPage renders correctly", () => {
    expect(domTree).toMatchSnapshot();
  });

  it('filter robots ', () => {
    const mockProps = {
      onRequestRobots: jest.fn(),
      robots: [],
      searchField: 'a',
      isPending: false
    }
    domTree = renderer.create(<MainPage {...mockProps}/>)
    expect(domTree.getInstance().filterRobots()).toEqual([]);
  });

  it('filter robots correctly ', () => {
    const filterRobots = [{
      id: 1,
      name: 'Leanne Graham',
      username: 'Bret',
      email: 'Sincere@april.biz'
    }]
    const mockProps = {
      onRequestRobots: jest.fn(),
      robots: [
        {
          id: 1,
          name: 'Leanne Graham',
          username: 'Bret',
          email: 'Sincere@april.biz'
        }
      ],
      searchField: 'a',
      isPending: false
    }
    domTree = renderer.create(<MainPage {...mockProps}/>)
    expect(domTree.getInstance().filterRobots()).toEqual(filterRobots);
  });

  it('is pending correctly to true', () => {
    const mockProps = {
      onRequestRobots: jest.fn(),
      robots: [],
      searchField: 'a',
      isPending: true
    }
    domTree = renderer.create(<MainPage {...mockProps}/>)
    expect(domTree.getInstance().filterRobots()).toEqual([]);
  });
});
