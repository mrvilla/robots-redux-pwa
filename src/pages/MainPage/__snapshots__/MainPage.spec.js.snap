// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Snapshot testing MainPage renders correctly 1`] = `
Array [
  <header
    className="tc mb4"
  >
    <h1>
      Robots
    </h1>
  </header>,
  <div
    className="tc mb4"
  >
    <div
      className="tc mb4"
    >
      <input
        className="pa2 ba"
        placeholder="Search robot"
        type="search"
      />
    </div>
  </div>,
  <div
    style={
      Object {
        "height": "800px",
        "overflow": "scroll",
      }
    }
  >
    <div
      className="tc"
    />
  </div>,
]
`;
